x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -170,7 +169,7 @@ export async function getPackageInfo({
       pkgVersion: resolvedPkgVersion,
       savedObjectsClient,
       installedPkg: savedObject?.attributes,
-      getPkgInfoFromArchive: packageInfo?.type === 'input',
+      getPkgInfoFromArchive: registryInfo?.type === 'input',
```

Comment:
Heads up that in [my PR ](https://github.com/elastic/kibana/pull/142353/files) I'm removing `getPkgInfoFromArchive` and relying only on the archive.